### PR TITLE
support ice restart by checking ufrag and pwd on incoming transport info

### PR DESCRIPTION
**Pull request by xdumaine**

### CODE REVIEW

**Comment by legastero** [comment]
I can't test at the moment, but I'm assuming your gateway is working with this patch, and it looks exactly like how we discussed.

 +1

**Comment by legastero** [comment]
Oh, wait. There's a callback parameter that needs to be called here, now that an offer/answer cycle is being used.

**Comment by legastero** [comment]
https://github.com/otalk/RTCPeerConnection/pull/70/files#diff-6acc65d68e969d4da0d7a1df72d4cde8L299

^ That `cb()` call needs to move up into the `else` clause, and a new `cb()` call made at the end of `processCandidates()`.

**Comment by xdumaine** [comment]
@legastero Good catch; fixed.

File: rtcpeerconnection.js [hunk 1]
```diff
@@ -254,24 +255,63 @@ PeerConnection.prototype.processIce = function (update, cb) {
...
+                        }
+                    );
+                    self._checkRemoteCandidate(iceCandidate);
+                });
+                cb();
+            };
+
+            if (self.config.remoteIce[content.name] && transport.ufrag &&
+                self.config.remoteIce[content.name].ufrag !== transport.ufrag) {
+                console.log('ice restart needed', transport);
```
Comment (fippo) [inline rtcpeerconnection.js]: uh, can you remove those console.logs? we should have no-console linting... but I only know how to do that for eslint.

File: rtcpeerconnection.js [hunk 2]
```diff
@@ -147,6 +147,7 @@ function PeerConnection(config, constraints) {
     this.config = {
         debug: false,
         ice: {},
+        remoteIce: {},
```
Comment (fippo) [inline rtcpeerconnection.js]: nit: I don't think this should be part of config. It is not configurable after all.

Comment (xdumaine) [inline rtcpeerconnection.js]: @fippo Good point - same for `config.ice` right?

Comment (fippo) [inline rtcpeerconnection.js]: yeah, that doesn't belong there either :-)

**Comment by fippo** [comment]
I don't see how you're actually triggering the ice restart here? You are only setting the remote description with new ufrag/pwd

**Comment by xdumaine** [comment]
@fippo setting remote description with a different ufrag triggers an ice restart automatically, correct? Or else something else is doing it magically...

**Comment by fippo** [comment]
You will need to call createAnswer and setLocalDescription. Just setting remote description is not enough

**Pull request merged**